lib/modules/platform/bitbucket/index.ts

```diff
@@ -179,7 +183,23 @@ export async function initRepo({
         )
       ).body
     );
-    config.defaultBranch = info.mainbranch;
+
+    mainBranch = info.mainbranch;
+
+    if (bbUseDevelopmentBranch) {
+      // Fetch Bitbucket development branch
+      developmentBranch = (
+        await bitbucketHttp.getJson<RepoBranchingModel>(
+          `/2.0/repositories/${repository}/branching-model`
+        )
+      ).body.development?.branch?.name;
+
+      if (developmentBranch) {
+        mainBranch = developmentBranch;
```

Comment:
this also means that the repo config is read from development branch. is that expected?

Reply:
Hmm, good call out.  

I just had a re-read of [this gitflow article](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) and it would seem to suggest that continuing to read the repo config from the "main branch" would make sense.

Is that what you takeaway too @viceice?  @rarkins how about you?

Reply:
I had simply assumed that it was your intent to read config from the development branch, so thanks @viceice for calling it out.

In that case it's a bit more complicated!

So we could, as this PR has done, use the development branch as _default_ branch in Renovate and read config + dependencies from there (assuming `baseBranches` is not configured and overrides that).

Alternatively, our behavior could be that `baseBranches=[developmentBranch]` unless `baseBranches` are explicitly configured by the user. In other words default branch, which config is read from, remains as-is. This explains why @viceice thought that this new option could be read from repo config while I thought it could not.

I'm ok with the second option

Reply:
Having slept on it - I'm leaning back towards keeping the implementation as-is and reading the configuration from the `develop` branch.  

Why?  this is the most active branch used for "integrating" feature branches.  it will be the one where developers are committing to most often.  the "main branch" is really just used for production releases.  If we were to read the renovate config from "main branch", that would mean teams that want to change their renovate configuration would have to do through a feature branch -> develop branch -> prod release workflow (assuming they dont commit directly to main 🤠 )